controllers/copy.go

```diff
@@ -39,6 +69,16 @@ func (r *StatefulSetReconciler) copyPVC(ctx context.Context, src client.ObjectKe
 		err = r.Client.Delete(ctx, &job, &client.DeleteOptions{
 			PropagationPolicy: &pol,
 		})
+		if r.SyncClusterRole != "" {
+			// Clean up RoleBinding
+			err = r.Client.Delete(ctx, &objs.RoleBinding, &client.DeleteOptions{
+				PropagationPolicy: &pol,
+			})
+			// Clean up ServiceAccount
+			err = r.Client.Delete(ctx, &objs.ServiceAccount, &client.DeleteOptions{
+				PropagationPolicy: &pol,
+			})
+		}
```

Comment:
There are two issues here:

* We completely ignore the errors. I know that problem was present before, but I just realized that. It the job deletion fails we should actually return an error. For the RoleBinding and ServiceAccount, just logging an error might be smarter.
* There is problem if multiple resize operations happen in the same namespace. The service account and rolebinding have a static name. So if two resize operations start: 
  - The first will create the SA and RB and start the job
  - The second will see that the SA and RB exists and start the job
  -  The first will finish, delete the SA and RB  -> Second job fails

  I see two ways we could fix this:

  * Choose a unique `RbacObjName` based on sts name
  * Simply don't delete the SA and RB.

Reply:
> There are two issues here:
> 
>     * We completely ignore the errors. I know that problem was present before, but I just realized that. It the job deletion fails we should actually return an error. For the RoleBinding and ServiceAccount, just logging an error might be smarter.

Done

> 
>     * There is problem if multiple resize operations happen in the same namespace. The service account and rolebinding have a static name. So if two resize operations start:
>       
>       * The first will create the SA and RB and start the job
>       * The second will see that the SA and RB exists and start the job
>       * The first will finish, delete the SA and RB  -> Second job fails
>       
>       I see two ways we could fix this:
>       
>       * Choose a unique `RbacObjName` based on sts name
>       * Simply don't delete the SA and RB.

The first approach was not very smart and would have created and deleted the SA and RoleBinding for each individual copy job. I've now refactored creating the RBAC objects directly into `resizeStatefulSet()` and only create the SA/RoleBinding pair at the start of the resize operation. The name is now generated to include the StatefulSet name to avoid collisions in situations where we have multiple StatefulSets in a single namespace.